nixos/modules/hardware/video/nvidia.nix

```diff
@@ -288,10 +294,8 @@ in
     environment.etc."egl/egl_external_platform.d".source =
       "/run/opengl-driver/share/egl/egl_external_platform.d/";
 
-    hardware.opengl.package = mkIf (!offloadCfg.enable) nvidia_x11.out;
-    hardware.opengl.package32 = mkIf (!offloadCfg.enable) nvidia_x11.lib32;
-    hardware.opengl.extraPackages = optional offloadCfg.enable nvidia_x11.out;
-    hardware.opengl.extraPackages32 = optional offloadCfg.enable nvidia_x11.lib32;
+    hardware.opengl.extraPackages = [ nvidia_x11.out ];
+    hardware.opengl.extraPackages32 = [ nvidia_x11.lib32 ];
```

Comment:
opengl.extraPackges should be used instead of overwriting the package because this causes many (if not all) of the mesa drivers to not be installed into `/run/opengl-driver/lib/dri/` this even includes things such as software renders.

Reply:
This should also match behavior for other distros where mesa will be available as well as nvidia libs once nvidia drivers are installed.

I also had to add libglvnd to extraPackages for something with multigpu but I didn't document why. >.>

Reply:
I'll see if libglvnd is needed.

Reply:
Maybe libglvnd would be helpful but I couldn't anything to support it as necessary so I don't think it should be included for now.